File: cpp/problem268.cpp
#include "mystd.h"
using namespace std;

/**
 * solution by getting the summation from 0 to n,
 *  and then subtracting each appeared one
 */
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int sum_0_to_n = (1+n)*n/2;
        for(auto num: nums){
            sum_0_to_n -= num;
        }
        return sum_0_to_n;
    }
};


int main(int argc, char const *argv[])
{
    // test with the example to Leetcode website
    vector<int> v1 = {9,6,4,2,3,5,7,0,1}, v2 = {3,0,1};
    Solution sol;
    cout << sol.missingNumber(v1) << " and " << sol.missingNumber(v2) << endl;
    return 0;
}

File: python/problem268.py
class Solution:
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        se=set([i for i in range(len(nums)+1)])
        se2=set(nums)
        return (se-se2).pop()
        
class Solution2:
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        while 1:
            try:
                if nums.pop()!=len(nums)+1:
                    return len(nums)+1
            except:
                return 0        